contracts/kernel/Kernel.sol

```diff
@@ -28,30 +31,54 @@ contract Kernel is IKernel, KernelStorage, Initializable, AppProxyFactory, ACLSy
         _setApp(APP_ADDR_NAMESPACE, ACL_APP_ID, acl);
 
         acl.initialize(_permissionsCreator);
+
+        // Hardocde constant to save gas
+        //recoveryVaultId = keccak256(APP_ADDR_NAMESPACE, apmNamehash("vault"));
+        recoveryVaultId = DEFAULT_VAULT_ID;
+    }
+
+    function newAppInstance(bytes32 _name, address _appBase) auth(APP_MANAGER_ROLE, arr(APP_BASES_NAMESPACE, _name)) public returns (ERCProxy appProxy) {
+        newAppInstance(_name, _appBase, false);
     }
 
     /**
     * @dev Create a new instance of an app linked to this kernel and set its base
     *      implementation if it was not already set
     * @param _name Name of the app
     * @param _appBase Address of the app's base implementation
+    * @param _setDefault Whether the app proxy app is the default one.
+    *        Useful when the Kernel needs to know of an instance of a particular app,
+    *        like Vault for escape hatch mechanism.
     * @return AppProxy instance
     */
-    function newAppInstance(bytes32 _name, address _appBase) auth(APP_MANAGER_ROLE, arr(APP_BASES_NAMESPACE, _name)) public returns (ERCProxy appProxy) {
+    function newAppInstance(bytes32 _name, address _appBase, bool _setDefault) auth(APP_MANAGER_ROLE, arr(APP_BASES_NAMESPACE, _name)) public returns (ERCProxy appProxy) {
         _setAppIfNew(APP_BASES_NAMESPACE, _name, _appBase);
         appProxy = newAppProxy(this, _name);
+        // By calling setApp directly and not the internal functions, we make sure the params are checked
+        // and it will only succeed if sender has permissions to set something to the namespace.
+        if (_setDefault) setApp(APP_ADDR_NAMESPACE, _name, appProxy);
```

Comment:
I love how this looks but we need to be consistent with the agreed style:

```solidity
if (x) {
  y()
}

Reply:
I'm fine with any standard you like, but I think it should be documented somewhere